benchmarks/transformer/score_mod.py

```diff
@@ -1077,59 +1135,365 @@ def generate_experiment_configs(
+        python score_mod.py -b 4 8 -s 1024 2048 -mods causal alibi --backend efficient
+
+    Args:
+        dynamic: Runs a dynamic shapes version of compiled flex attention
+        calculate_bwd: Calculate backward pass times
+        dtype: Data type for tensors (bfloat16, float16, float32)
+        b: Batch sizes to benchmark
+        nh: Number of query and key/value heads in format "Hq,Hkv"
+        s: Sequence lengths to benchmark
+        d: Head dimensions to benchmark
+        mods: Score modifications: noop, causal, rel, head_bias, alibi, sliding_window, document_mask, prefix_lm, softcap
+        backend: Backends for attention computation: math, efficient, cudnn, fav2, fav3, fakv, og-eager
+        max_autotune: Turn on max-autotune optimization
+        decoding: Benchmark decoding mode (query sequence length = 1)
+        kv_size: Key/value cache size in MiB (ignores batch size if specified)
+        throughput: Calculate kernel memory bandwidth & computational throughput (always True)
+        save_path: Path to save the results CSV file
+        output_json_for_dashboard: Path to save results in JSON format for PyTorch OSS dashboard
+        benchmark_name: Name of the benchmark for dashboard output
+    """
+    # Convert dtype string to torch dtype (if not already converted)
+    import torch
+
+    if isinstance(dtype, str):
+        dtype = getattr(torch, dtype)
+
+    # Parse head configurations (if not already parsed)
+    nh_parsed = []
+    for h in nh:
+        if isinstance(h, tuple):
+            nh_parsed.append(h)
+        else:
+            nh_parsed.append(heads_input_type(h))
+
+    # Always calculate throughput
+    throughput = True
+    print('Backend: ', backend)
     seed = 123
     np.random.seed(seed)
     torch.manual_seed(seed)
     results = []
+    experiment_count = 0
     for config in tqdm(
         generate_experiment_configs(
-            args.calculate_bwd,
-            args.dtype,
-            args.b,
-            args.nh,
-            args.s,
-            args.d,
-            args.mods,
-            args.decoding,
-            args.kv_size,
-            args.throughput,
-            args.backend,
+            calculate_bwd,
+            dtype,
+            b,
+            nh_parsed,
+            s,
+            d,
+            mods,
+            decoding,
+            kv_size,
+            throughput,
+            backend,
+            max_autotune,
         )
     ):
         results.append(
             Experiment(
                 config,
                 run_single_experiment(
                     config,
-                    dynamic=args.dynamic,
-                    max_autotune=args.max_autotune,
+                    dynamic=dynamic,
                 ),
             )
         )
 
-    print_results(results, args.save_path)
+        experiment_count += 1
+        # Periodic memory cleanup every 10 experiments
+        if experiment_count % 10 == 0:
+            cleanup_memory()
```

Comment:
what's the reason behind manually doing memory cleanup? I expect this to be handled automatically, is this not the case?